js/apps/admin-ui/cypress/support/util/AdminClient.ts

```diff
@@ -149,6 +158,30 @@ class AdminClient {
     });
   }
 
+  async addClientRoleToUser(
+    userId: string,
+    clientId: string,
+    roleNames: string[],
+  ) {
+    await this.#login();
+
+    const client = await this.#client.clients.find({ clientId });
+    const clientRoles = await Promise.all(
+      roleNames.map(
+        async (roleName) =>
+          (await this.#client.clients.findRole({
+            id: client[0].id!,
+            roleName: roleName,
+          })) as RoleMappingPayload,
+      ),
+    );
+    await this.#client.users.addClientRoleMappings({
+      id: userId,
+      clientUniqueId: client[0].id!,
+      roles: clientRoles,
+    });
```

Comment:
I think this could be slightly simplified by using `findOne()`
```suggestion
    const client = await this.#client.clients.findOne({ id: clientId });
    const clientRoles = await Promise.all(
      roleNames.map(
        async (roleName) =>
          (await this.#client.clients.findRole({
            id: client!.id!,
            roleName: roleName,
          })) as RoleMappingPayload,
      ),
    );
    await this.#client.users.addClientRoleMappings({
      id: userId,
      clientUniqueId: client!.id!,
      roles: clientRoles,
    });
```